app/services/scrape.rb

```diff
@@ -1,117 +1,237 @@
 class Scrape
 
+  require'resolv'
+  # require 'selenium-webdriver'
+
   module Tags
     SCRIPT = 'script'
-    LINK = 'links'
+    LINK = 'link'
+  end
+
+  module SubSource
     SRC = 'src'
     HREF = 'href'
   end
 
   module DataTypes
     PLUGINS = 'plugins'
+    MUPLUGINS = 'mu-plugins'
     THEMES = 'themes'
     JS = 'js'
     CLOUDFLARE = 'cloudflare'
   end
 
-
-  def self.filter_wp_urls(urls, logger)
+  def self.filter_wp_urls(urls, logger, test_id)
     url_html_version_map = Hash.new{ |h,k| h[k] = Hash.new }
     threads = []
-    logger.info "fetching file"
-    file_data = File.readlines(File.join(__dir__,'proxy.txt'), chomp: true)
-    puts file_data.count
-    for i in 0..5 do
-      puts file_data[i]
-      urls.each do |url|
-        puts url
-        threads << Thread.new(){
-          thread_block(url,url_html_version_map,file_data[i])
-        }
-      end
+    # _proxy = ProxyDatum.order('RANDOM()').first
+    urls.each do |url_id|
+      threads << Thread.new(){
+        thread_block(url_id, url_html_version_map, logger, test_id)
+      }
     end
     threads.each do |thread|
       thread.join
     end
     return url_html_version_map
   end
 
-  def self.thread_block(url, url_html_version_map,proxy_ip)
+  def self.thread_block(url_id, url_html_version_map, logger, test_id)
     begin
-      puts url + "    " + proxy_ip
-      RestClient.proxy = "http://" + proxy_ip
-      html = Nokogiri::HTML.parse(RestClient.get url)
-      _url = Url.where(url: url).first
-      if _url 
-        _url_id = _url.id
-        url_html_version_map[_url_id] = {:html => html, :version => _url.site_data_info.cms_version}
-      else
-        _version = check_wordpress_in_meta(html)
-        if _version
-          _url_id =  Url.import_urls([url])[0]
-          url_html_version_map[_url_id] = {:html => html, :version => _version}
-        end
+      url = Url.find(url_id)
+      html = Nokogiri::HTML.parse(RestClient.get (url.url + "?x=#{rand(999999)}"))
+      #File.write("/tmp/#{url.url}", html)
+      cms_and_version_hash = cms_and_version(html)  # fetching both cms type and its version together 
+      if cms_and_version_hash.present?
```

Comment:
`if cms_and_version(html).present?`

Reply:
we are using cms_and_version returned object in the if statement so why compute it again